cli/tests/pcluster/validators/test_all_validators.py

```diff
@@ -295,3 +309,296 @@ def test_validators_are_called_with_correct_argument(test_datadir, mocker):
     ebs_volume_size_snapshot_validator.assert_called()
     shared_ebs_volume_id_validator.assert_called()
     fsx_persistent_options_validator.assert_called()
+
+
+def test_scheduler_plugin_all_validators_are_called(test_datadir, mocker):
+    """Verify that all validators are called during validation."""
```

Comment:
Would it get too ugly to generalize the already existing `test_all_validators_are_called()` and `test_validators_are_called_with_correct_argument()` adding a new `scheduler` parameters in the signature, which can then be controlled via a `pytest.mark.parametrize` annotation? This way it would be easier to see any difference in the validation between "normal scheduler" and scheduler plugin. Or is it really better to split the two sets of validators?

Reply:
I think it is better to split the two sets of validators. The reason is that: The cluster configurations are different, for example, you will see 
`    instance_architecture_compatibility_validator.assert_has_calls(
        [
            call(instance_type="t2.large", architecture="x86_64"),
            call(instance_type="c4.2xlarge", architecture="x86_64"),
            call(instance_type="c5.4xlarge", architecture="x86_64"),
            call(instance_type="c5d.xlarge", architecture="x86_64"),
            call(instance_type="t2.large", architecture="x86_64"),
        ]
    )`
in slurm.yaml and scheduler_plugin.yaml Although they are using same validators `instance_architecture_compatibility_validator`, they have different numbers of calls and also different parameter values. If we use pytest.mark.parametrize, there are lots of validators in the test, we will need to mark all these validators as parameters. Each time we add a new validator, we need to add a parameter in `pytest.mark.parametrize`, it is not easy to maintain.
What is more, there are some validators differences for slurm scheduler and plugin scheduler. Then we will add if... else...to check which is the scheduler if we put it in one test. IMO it is easier to add validators and test cases when in the future we add more validators for slurm and scheduler plugin.